docs/reference/modules/ROOT/pages/query-configuration.adoc

```diff
@@ -0,0 +1,86 @@
+= Query Engine Configuration
+
+There are a number of different defaults that can be overriden within the Crux query engine.
+
+== Overriding Defaults
+
+We can override the arguments of `:crux/query-engine`.
+
+[tabs]
+====
+JSON::
++
+[source,json]
+----
+{
+  "crux/query-engine": {
+    "query-timeout": 60000
+  }
+}
+----
+
+Clojure::
++
+[source,clojure]
+----
+{:crux/query-engine {:query-timeout 60000}}
+----
+
+EDN::
++
+[source,clojure]
+----
+{:crux/query-engine {:query-timeout 60000}}
+----
+====
+
+== Parameters
+
+* `entity-cache-size` (int, default 32768): query entity cache size.
+* `query-timeout` (int, default 30000): query timeout in milliseconds.
+* `batch-size` (int, default 100): batch size of results.
+* `fn-allow-list` (xref:#fn-allowlist[`Predicate Allowlist`], default nil): list of allowed namespaces/functions in predicate functions.
+
+[#fn-allowlist]
+== Predicate Allowlist
+
+The query engine can be provided with an allowlist to configure which predicate functions can be ran inside of queries. By default, any `clojure.core` or fully quantified function name is allowed within a query, though there are many reasons to make this restrictive.
```

Comment:
This doesn't read quite right to me, think it's mainly the use of the passive voice which makes it a little wordy. I'm aiming for the voice of the Reference section to read a lot more like a specification rather than a tutorial - authoritative, terse, dry, boring almost. Something like this?

```
By default, users can invoke any predicate function in their queries.
To restrict what functions a user can invoke, you can specify a `:fn-allow-list` of safe functions/namespaces.
We include a set of pure functions from `clojure.core`, for convenience.
```

I'm also assured that the adoc idiom is one sentence per line :)